app/src/main/java/com/example/bottomnav/DataDecoder.java

```diff
@@ -0,0 +1,56 @@
+package com.example.bottomnav;
+
+public interface DataDecoder {
+    static DataDecoder getPrimativeDecoder(PayLoadDataType dataType) {
```

Comment:
I'd prefer this returned an Optional<DataDecoder> to explicitly indicate that it can fail